extensions/wikia/EmergencyBroadcastSystem/EmergencyBroadcastSystemController.class.php

```diff
@@ -64,4 +65,44 @@ protected function hasNonPortableInfoBoxes() {
 		// TODO: Actually check for non portable infoboxes
 		return true;
 	}
+
+	protected function buildSurveyUrl() {
+		global $wgContLang;
+
+		$context = $this->getContext();
+		$userLanguage = $context->getLanguage();
+		$userName = $context->getUser()->getName();
+
+		$url = $this->googleFormUrl( $userLanguage->mCode, $userName );
+
+		if ( $url === null ) {
+			$url = $this->googleFormUrl( $wgContLang->mCode, $userName );
+		}
+
+		if ( $url === null ) {
+			$url = $this->googleFormUrl( 'en', $userName );
+		}
+
+		return $url;
+	}
+
+	protected function googleFormUrl( $language, $userName ) {
```

Comment:
So in this approach the array below is going to be defined up to three times, I don't think there is a good reason for that.
Also, isn't accessing not defined index of an array throwing a notice?